ciphers/base32.py

```diff
@@ -1,42 +1,49 @@
-import base64
+"""
+Base32 encoding and decoding
 
+https://en.wikipedia.org/wiki/Base32
+"""
+B32_CHARSET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567"
 
-def base32_encode(string: str) -> bytes:
+
+def base32_encode(data: bytes) -> bytes:
     """
-    Encodes a given string to base32, returning a bytes-like object
-    >>> base32_encode("Hello World!")
+    >>> base32_encode(b"Hello World!")
     b'JBSWY3DPEBLW64TMMQQQ===='
-    >>> base32_encode("123456")
+    >>> base32_encode(b"123456")
     b'GEZDGNBVGY======'
-    >>> base32_encode("some long complex string")
+    >>> base32_encode(b"some long complex string")
     b'ONXW2ZJANRXW4ZZAMNXW24DMMV4CA43UOJUW4ZY='
     """
-
-    # encoded the input (we need a bytes like object)
-    # then, b32encoded the bytes-like object
-    return base64.b32encode(string.encode("utf-8"))
+    binary_data = "".join([bin(ord(d))[2:].zfill(8) for d in data.decode("utf-8")])
```

Comment:
Can we use a generator rather than a list?
```suggestion
    binary_data = "".join(bin(ord(d))[2:].zfill(8) for d in data.decode("utf-8"))
```